src/components/checkbox/checkbox.ts

```diff
@@ -248,6 +215,35 @@ export class MdCheckbox implements ControlValueAccessor {
     }
   }
 
+  /*
+   * We use a hidden native input field to handle changes to focus state via keyboard navigation,
+   * with visual rendering done separately. The native element is kept in sync with the overall
+   * state of the component.
+   */
+  onInputFocus() {
+    this.hasFocus = true;
+  }
+
+  onInputBlur() {
+    this.hasFocus = false;
```

Comment:
Good catch- that `(blur)` on the host should be removed and the `onTouched` should be called here.